Mount /dev/pts with proper mode and group. Without this, performing a
local build will subsequently cause grantpt(3) errors for users. This
can be seen in gnome-terminal failing to open new windows or tabs.

This is a backport of
https://github.com/openSUSE/obs-build/commit/629be005

--- a/build
+++ b/build
@@ -330,7 +330,7 @@ setup_virtual_filesystems()
     mkdir -p $BUILD_ROOT/dev/pts
     mkdir -p $BUILD_ROOT/dev/shm
     mount -n -tproc none $BUILD_ROOT/proc || true
-    mount -n -tdevpts none $BUILD_ROOT/dev/pts
+    mount -n -tdevpts -omode=0620,gid=5 none $BUILD_ROOT/dev/pts
     if ! test -f $BUILD_ROOT/dev/shm/.mounted ; then
       mount -n -ttmpfs none $BUILD_ROOT/dev/shm
       touch $BUILD_ROOT/dev/shm/.mounted
--- a/init_buildsystem
+++ b/init_buildsystem
@@ -113,7 +113,7 @@ setup_virtual_filesystems()
     mkdir -p $BUILD_ROOT/dev/pts
     mkdir -p $BUILD_ROOT/dev/shm
     mount -n -tproc none $BUILD_ROOT/proc || true
-    mount -n -tdevpts none $BUILD_ROOT/dev/pts
+    mount -n -tdevpts -omode=0620,gid=5 none $BUILD_ROOT/dev/pts
     if ! test -f $BUILD_ROOT/dev/shm/.mounted ; then
       mount -n -ttmpfs none $BUILD_ROOT/dev/shm
       touch $BUILD_ROOT/dev/shm/.mounted
